models/ConsumerCoupon.js

```diff
@@ -0,0 +1,30 @@
+const mongoose = require("mongoose");
+
+const ConsumerCouponSchema = new mongoose.Schema(
+  {
+    consumerId: {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: "Consumer",
+      required: true,
+    },
+    couponId: {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: "Coupon",
+      required: true,
+    },
+    qrCode: {
+      type: String,
+      required: true,
+    },
+    status: {
+      type: String,
+      required: true,
+      enum: ["active", "removed", "expired"],
+    },
```

Comment:
add a qrIdentification attribute that stores a six-digit code that relates to the dataURL of the QR. This serves as a fallback when the qr scanner fails.